asynciterator.ts

```diff
@@ -1251,6 +1259,104 @@ function destinationFillBuffer<S>(this: InternalSource<S>) {
     (this._destination as any)._fillBuffer();
 }
 
+interface ComposedFunction {
+  fn: Function,
+  close?: boolean,
+  next?: ComposedFunction
+}
+
+export class MappingIterator<T, D = T> extends AsyncIterator<D> {
+  private _fn?: Function;
+  private _destroySource: boolean;
+  private onSourceError = (err: Error) => this.emit('error', err);
+  private onSourceReadable = () => this.emit('readable');
+  private onSourceEnd = () => this.close();
+
+  get readable() {
+    return this.source.readable;
+  }
+
+  set readable(readable) {
+    this.source.readable = readable;
+  }
+
+  constructor(
+    protected source: AsyncIterator<T>,
+    private transforms?: ComposedFunction,
```

Comment:
I'd still just pass `mapping: Function | Array<Function>`; I think the overhead of creating arrays for this purpose is the same than creating the linked list. Note that the array is never manipulated after creation, so that's a very different case than the one we had.